### PR TITLE
Remove inappropriate link in API doc for LocationStrategy

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
Removes automatic link from example URL string.

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/angular/angular/issues/31012

## What is the new behavior?
text is in code font with no link

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/common/src/location/location.ts [hunk 1]
```diff
@@ -43,7 +43,8 @@ export interface PopStateEvent {
  *
  * ### Example
  *
- * {@example common/location/ts/path_location_component.ts region='LocationComponent'}
+ * <code-example path="common/location/ts/path_location_component.ts"
+ * region='LocationComponent'><?code-example>
```
Comment (gkalpak) [inline packages/common/src/location/location.ts]: - `<?code-example>` --> `</code-example>`
- It is better to use consistent quotes: `region='...'` --> `region="..."`

File: packages/common/src/location/location_strategy.ts [hunk 2]
```diff
@@ -12,16 +12,17 @@ import {LocationChangeListener} from './platform_location';
...
  *
- * Applications should use the {@link Router} or {@link Location} services to
+ * Applications should use the `Router` or `Location` services to
  * interact with application route state.
  *
- * For instance, {@link HashLocationStrategy} produces URLs like
- * `http://example.com#/foo`, and {@link PathLocationStrategy} produces
- * `http://example.com/foo` as an equivalent URL.
+ * For instance, `HashLocationStrategy` produces URLs like
+ * <code class="no-auto-link">http://example.com#/foo</code>,
```
Comment (gkalpak) [inline packages/common/src/location/location_strategy.ts]: FWIW, this won't be necessary once #31051 lands.

**Comment by AndrewKushnir** [comment]
@jbogarthyde @gkalpak since PR #31051 is close to be merged, could you plz confirm if anything should be updated in this PR (based on the comment from George)? I will add the "cleanup" label for now to avoid this PR from being merge accidentally. Thank you.

**Comment by jbogarthyde** [comment]
This makes a few minor editorial/style changes as well.

The no-code link will still work even with George's fix -- it removes the unwanted link to the HTTP doc, but makes the entire URL a link to an example page, which I think is actually kind of nicer and more readable than straight code font with no link at all.

**Pull request closed without merging**